controllers/storagecluster/storageconsumer.go

```diff
@@ -160,6 +189,40 @@ func (s *storageConsumer) ensureDeleted(r *StorageClusterReconciler, storageClus
 	return ctrl.Result{}, nil
 }
 
+func getInternalStorageConsumerName(r *StorageClusterReconciler, storageClientName, namespace string) (string, error) {
+	storageConsumerObj := &ocsv1a1.StorageConsumer{}
+	storageConsumerObj.Name = defaults.LocalStorageConsumerName
+	storageConsumerObj.Namespace = namespace
+	if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(storageConsumerObj), storageConsumerObj); client.IgnoreNotFound(err) != nil {
+		return "", fmt.Errorf("failed to get the internal storageconsumer: %v", err)
+	}
+	if storageConsumerObj.UID == "" {
+		storageClientObj := &ocsclientv1a1.StorageClient{}
+		storageClientObj.Name = storageClientName
+		if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(storageClientObj), storageClientObj); client.IgnoreNotFound(err) != nil {
+			return "", fmt.Errorf("failed to get the internal storageclient: %v", err)
+		}
+		if storageClientObj.UID != "" && storageClientObj.Status.ConsumerID != "" {
+			// Get the StorageConsumer from UID
+			storageConsumers := &ocsv1a1.StorageConsumerList{}
+			if err := r.Client.List(
+				r.ctx,
+				storageConsumers,
+				client.MatchingFields{util.UIDIndexName: storageClientObj.Status.ConsumerID},
+			); err != nil {
+				return "", fmt.Errorf("failed to list the internal storageconsumers: %v", err)
+			}
+			if len(storageConsumers.Items) > 1 {
+				return "", fmt.Errorf("more than one storage consumer is available for the storage client")
+			}
+			if len(storageConsumers.Items) == 1 {
```

Comment:
can you club the condition and move the high chance to first if?
`if len == 1 { return first } else if len > 1 { return err }`